Application/Query Handlers/GetUsersQueryHandler.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Application.Queries;
+using Domain.Entities;
+using MediatR;
+using Persistence.Repository_Interfaces;
+
+namespace Application.Query_Handlers
+{
+    public class GetUsersQueryHandler : IRequestHandler<GetAllUsersQuery, IEnumerable<ApplicationUser>>
+    {
+        private readonly IUserRepository _userRepository;
+
+        public GetUsersQueryHandler(IUserRepository userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        public async Task<IEnumerable<ApplicationUser>> Handle(GetAllUsersQuery request,
+            CancellationToken cancellationToken)
+        {
+            var users = _userRepository.GetUsersAsync();
+                
+            if (users == null) return null;
```

Comment:
Please check your code, you try to check nullable for Task object. IT IS SUPER USELESS
Just add await for 23 line of code

Reply:
Done